File: 1-250/73. Set Matrix Zeroes/solution.py
from typing import List

class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        n = len(matrix)
        m = len(matrix[0])

        col1 = False
        row1 = False
        for i in range(m):
            if matrix[0][i] == 0:
                row1 = True
                break
        for i in range(n):
            if matrix[i][0] == 0:
                col1 = True
                break
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == 0:
                    matrix[0][j] = 0
                    matrix[i][0] = 0

        for i in range(1, n):
            for j in range(1, m):
                if matrix[0][j] == 0 or matrix[i][0] == 0:
                    matrix[i][j] = 0

        if row1:
            for i in range(m):
                matrix[0][i] = 0
        if col1:
            for i in range(n):
                matrix[i][0] = 0

File: 1-250/73. Set Matrix Zeroes/solution.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int n  = matrix.length ; 
        int m = matrix[0].length ; 
        boolean[] rows = new boolean[n] ; 
        boolean[] cols = new boolean[m] ; 
        for( int i=0 ; i<n ; ++i ){
            for( int j=0 ; j<m ; ++j ){
                if( matrix[i][j] == 0  ){
                    rows[i] = true ; 
                    cols[j] = true ; 
                }
            }
        }   
        for( int i=0 ; i<n ; ++i ){
            if( rows[i] ){
                setRow( matrix , i ) ; 
            }
        }
        for( int i=0 ; i<m ; ++i ){
            if( cols[i] ){
                setCol( matrix , i ) ; 
            }
        }

    }
    public void setRow( int[][] arr , int i  ){
        for( int t=0 ; t<arr[0].length ; ++t ){
                arr[i][t] = 0 ;             
        }
    }
    public void setCol( int[][] arr , int j  ){
        for( int t=0 ; t<arr.length ; ++t ){
                arr[t][j] = 0 ; 
        }
    }
}
